pseudomonas_analysis/nbconverted/viz_expression_data.py

```diff
@@ -0,0 +1,579 @@
+
+
+# Plot
+
+# Drop label column
+normalized_all_data_numeric = normalized_all_data.drop(['sample group'], axis=1)
+
+model = umap.UMAP(random_state=1).fit(normalized_all_data_numeric)
+
+normalized_all_data_UMAPencoded = model.transform(normalized_all_data_numeric)
+normalized_all_data_UMAPencoded_df = pd.DataFrame(data=normalized_all_data_UMAPencoded,
+                                         index=normalized_all_data.index,
+                                         columns=['1','2'])
+
+# Add back label column
+normalized_all_data_UMAPencoded_df['sample group'] = normalized_all_data['sample group']
+
+# Plot
+fig = pn.ggplot(normalized_all_data_UMAPencoded_df, pn.aes(x='1', y='2'))
+fig += pn.geom_point(pn.aes(color='sample group'), alpha=0.4)
+fig += pn.labs(x ='UMAP 1',
+            y = 'UMAP 2',
+            title = 'Gene expression data in gene space')
+fig += pn.theme_bw()
+fig += pn.theme(
+    legend_title_align = "center",
+    plot_background=pn.element_rect(fill='white'),
+    legend_key=pn.element_rect(fill='white', colour='white'), 
+    legend_title=pn.element_text(family='sans-serif', size=15),
+    legend_text=pn.element_text(family='sans-serif', size=12),
+    plot_title=pn.element_text(family='sans-serif', size=15),
+    axis_text=pn.element_text(family='sans-serif', size=12),
+    axis_title=pn.element_text(family='sans-serif', size=15)
+    )
+fig += pn.scale_color_manual(['#bdbdbd', 'red', 'blue'])
+fig += pn.guides(colour=pn.guide_legend(override_aes={'alpha': 1}))
+
+fig += pn.scales.xlim(9,10)
+print(fig)
+
+
+# Based on a UMAP of the normalized gene expression data, it looks like there isn't a clear separation between WT and mutant samples, though there are only 2 samples per group so this type of clustering observation is limited.
+# 
+# **Takeaway:**
+# 
+# In trying to understand why there are these flat-tops to some of the volcano plots and why some volcano plots are completely flat, we found:
+# 1. This behavior is _not_ a result of how we are plotting in python (there was some speculation about there being an issue with the numpy library used)
+# 2. The latent space shifting we're doing seems to roughly preserve differences between groups, but this signal can be muddled/noisy depending on where the experiment was shifted to (i.e. the representation that is found in that location can cause the experiment to have a more compressed difference between groups) as seen in the heatmaps. The heatmap of the two simulation experiments shows that some experiments have a more noisey distinction between groups (WT vs mutant) whereas the other simulation experiment has a more distinct difference where the within grouping is cleaner. This definitely points to the need to understand how this simulation process is working and how biology is represented in the latent space. This will definitely be a project for the future. For now we at least have an explanation for why we are observing these shapes in the volcano plots
```

Comment:
>  The latent space shifting we're doing seems to roughly preserve differences between groups

Is this based on the heatmap for the top DE genes? I wonder if there's some way you can quantitatively show the difference (you don't have to answer this question now, just something to think about for the future)

Reply:
This is a good point. I will make sure to update the text here. The reason we're claiming that the differences are preserved roughly is from the previous publication showing that heatmaps between simulated and template experiments had a consistent sample structure but there was a different set of related groups of genes that were DE indicating that the simulated experiment represented a new real experiment.